Added routers and onboarding screen

tsumith opened the pull request:
added app_router (uses go_router) into config for handling navigation.
added onboarding into features for testing.
added images into assets for onboarding images.

maniraj1234 commented on frontend/lib/assets/images/img1.png:
give readable names to the images.

tsumith replied:
modified names of the images as :
onboarding_chat
onboarding_video_call
onboarding_connect

maniraj1234 commented on frontend/lib/config/.env:
can you create a template.env file.

ideally all the keys should be present in template.env without values.

tsumith replied:
created 'template.env' and added ideal keys that are required.

maniraj1234 commented on frontend/lib/config/app_router.dart:
```diff
@@ -0,0 +1,32 @@
+);
+
+///For undefined routes return a [ErrorScreen]
+class ErrorScreen extends StatelessWidget {
```
modulairze this.

tsumith replied:
created separate files for handling errors and error views

maniraj1234 commented on frontend/lib/config/app_router.dart:
```diff
@@ -0,0 +1,32 @@
+import 'package:go_router/go_router.dart';
+
+///[router] serves as main router
+final GoRouter router = GoRouter(
```
https://dart.dev/effective-dart/documentation

e.g:
/// Defines a flag with the given [name] and [abbreviation].
///
/// The [name] and [abbreviation] strings must not be empty.
///
/// Returns a new flag.
///
/// Throws a [DuplicateFlagException] if there is already an option named
/// [name] or there is already an option using the [abbreviation].
Flag addFlag(String name, String abbreviation) => ...

tsumith replied:
added doc comments as following the dart documentation.

maniraj1234 commented on frontend/lib/features/onboarding/onboarding.dart:
```diff
@@ -0,0 +1,142 @@
+  int _currentPage = 0;
+
+  ///onboarding screen data
+  final List<Map<String, String>> _onboardingData = [
```
add this to a json. read it from there.

tsumith replied:
separated the data by creating 'onboarding_data.json' under assets/json.

maniraj1234 commented on frontend/lib/main.dart:
```diff
@@ -1,122 +1,22 @@
-      _counter++;
-    });
-  }
+class MyApp extends StatelessWidget {
```
give a meaningful name;
~MyApp~  
can be RootWidget, RootApp, MainScreen, MainApp, App or anything else

tsumith replied:
renamed the app widget as AppRoot.

maniraj1234 commented on frontend/lib/core/navigation/main_router.dart:
````diff
@@ -0,0 +1,39 @@
+///   routerConfig: router,
+/// );
+/// ```
+///
````
nit; remove these blank line commments

maniraj1234 commented on frontend/lib/core/navigation/main_router.dart:
```diff
@@ -0,0 +1,39 @@
+
+  /// **Application Routes**
+  ///
+  /// - [`OnboardingScreen`] serves as the main view (for testing purposes).
```
for testing purposes?
wouldn't it be this later?

tsumith replied:
OnboardingScreen should be shown only once, when the user installs the app for the first time. So, we might need to modify it later and add separate logic for that. we might add splash screen too.

maniraj1234 commented on frontend/lib/features/onboarding/onboarding.dart:
```diff
@@ -0,0 +1,178 @@
+/// This is then first screen that users see when they open the app.
+/// Provides a series of onboarding pages that introduce the app's features.
+/// Presents swipable [pageView] pages
+
```
nit; remove blank line

maniraj1234 commented on frontend/lib/features/onboarding/onboarding.dart:
```diff
@@ -0,0 +1,178 @@
+  State<OnboardingScreen> createState() => _OnboardingScreenState();
+}
+
+class _OnboardingScreenState extends State<OnboardingScreen> {
```
can we decouple the screen and screenstate into 2 files?

maniraj1234 commented on frontend/lib/main.dart:
```diff
@@ -1,122 +1,26 @@
+  ///
+  ///
+  ///load environment variables from .env file
+  // await dotenv.load(fileName: "lib/config/.env");
```
why is this commented?

maniraj1234 commented on frontend/lib/main.dart:
```diff
@@ -1,122 +1,26 @@
-      _counter++;
-    });
-  }
+class AppRoot extends StatelessWidget {
```
Create another component for AppRoot

tsumith commented:
feat: Added onboarding screen and router setup

- Implemented onboarding UI with JSON data
- Configured main router for navigation
- Added error handling for undefined routes
![onboarding_chat](https://github.com/user-attachments/assets/6f34a042-3af4-4b15-9846-63f76ec88ef3)
![onboarding_video_call](https://github.com/user-attachments/assets/cc85dd39-52a9-4f80-bf13-aa18bebaa253)
![onboarding_connect](https://github.com/user-attachments/assets/bbe510a2-0463-4da6-9a2f-933c56be4d3b)
